kernel/src/loader/mod.rs

```diff
@@ -229,51 +322,93 @@ mod tests {
         assert_eq!(val, b'\0');
     }
 
+    #[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
+    #[test]
     // Elf64 image that prints hello world on x86_64.
-    fn make_elf_bin() -> Vec<u8> {
-        let mut v = Vec::new();
-        v.extend_from_slice(include_bytes!("test_elf.bin"));
-        v
+    fn test_load_kernel() {
+        let gm = create_guest_mem();
+        let image = make_test_bin();
+        assert_eq!(
+            Ok(GuestAddress(0x10_0000)),
+            load_kernel(&gm, &mut Cursor::new(&image), 0)
+        );
     }
 
+    #[cfg(target_arch = "aarch64")]
     #[test]
-    fn load_elf() {
+    // aarch64 kernel image with specific header.
+    fn test_load_kernel() {
         let gm = create_guest_mem();
-        let image = make_elf_bin();
+        let image = make_test_bin();
         assert_eq!(
-            Ok(GuestAddress(0x10_0000)),
+            Ok(GuestAddress(0x80000)),
             load_kernel(&gm, &mut Cursor::new(&image), 0)
         );
     }
 
+    #[cfg(target_arch = "aarch64")]
+    #[test]
+    fn load_kernel_no_memory() {
+        let gm = GuestMemory::new(&[(GuestAddress(0x0), 79)]).unwrap();
+        let image = make_test_bin();
+        assert_eq!(
+            Err(Error::ReadKernelImage),
+            load_kernel(&gm, &mut Cursor::new(&image), 0)
+        );
+    }
+
+    #[cfg(target_arch = "aarch64")]
+    #[test]
+    fn test_load_bad_kernel() {
+        let gm = create_guest_mem();
+        let mut bad_image = make_test_bin();
+        bad_image.truncate(56);
+        assert_eq!(
+            Err(Error::ReadProgramHeader),
+            load_kernel(&gm, &mut Cursor::new(&bad_image), 0)
+        );
+
+        bad_image.truncate(4);
```

Comment:
Since we're getting the same error as before, does it make sense to truncate further to 4 and test again here?